### PR TITLE
Draft: docker config inheritance (chaining)

### PR DESCRIPTION
**Pull request by PiperS52**
## Changes in this PR

- refactoring docker-compose files to extract out common tasks and extend services

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -45,7 +45,7 @@
     t.datetime "created_at", precision: 6, default: -> { "CURRENT_TIMESTAMP" }, null: false
     t.datetime "updated_at", precision: 6, default: -> { "CURRENT_TIMESTAMP" }, null: false
     t.integer "journeys_count"
-    t.string "slug", null: false
+    t.string "slug"
```
Comment (peterdavidhamilton) [inline db/schema.rb:48]: Also, watch out for committing unrelated files.

**Comment by peterdavidhamilton** [comment]
Can you push what you have most recently please

File: .github/workflows/continuous-integration.yml [hunk 2]
```diff
@@ -27,9 +27,9 @@ jobs:
         run: docker load -i /tmp/docker-save/snapshot.tar || true
         if: steps.cache-docker.outputs.cache-hit == 'true'
       - name: Build
-        run: docker-compose -f docker-compose.ci.yml -p app build
+        run: docker-compose -f docker-compose.yml -f docker-compose.test.yml -f docker-compose.ci.yml -p app build
       - name: Test
-        run: docker-compose -f docker-compose.ci.yml run --rm test
+        run: docker-compose -f docker-compose.yml -f docker-compose.test.yml -f docker-compose.ci.yml run --rm app
```
[the remark this answers is not in the document]
Comment (PiperS52) [reply .github/workflows/continuous-integration.yml]: This suggestion has been actioned in the latest version within the 'dev/test gems in production' PR after merging this branch

**Comment by peterdavidhamilton** [comment]
[Docker config inheritance (chaining)](https://trello.com/c/6RtcvDDF/39-docker-config-inheritance-chaining)

**Comment by peterdavidhamilton** [comment]
Safe to close when you're ready.
